packages/autocomplete-js/src/__tests__/render.test.ts

```diff
@@ -142,8 +432,140 @@ describe('render', () => {
       expect(
         panelContainer.querySelector<HTMLElement>('.aa-Panel')
       ).toBeInTheDocument();
+    });
+  });
 
-      expect(panelContainer).toHaveTextContent('testSource');
+  test('provides a default Fragment', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+
+    document.body.appendChild(panelContainer);
+    autocomplete<{ label: string }>({
+      container,
+      panelContainer,
+      openOnFocus: true,
+      getSources() {
+        return [
+          {
+            sourceId: 'testSource',
+            getItems() {
+              return [{ label: '1' }];
+            },
+            templates: {
+              item({ item }) {
+                return item.label;
+              },
+            },
+          },
+        ];
+      },
+      render({ createElement, Fragment }, root) {
+        expect(Fragment).toBe(PreactFragment);
+        render(createElement(Fragment, null, 'testSource'), root);
+      },
+    });
+
+    const input = container.querySelector<HTMLInputElement>('.aa-Input');
+
+    fireEvent.input(input, { target: { value: 'a' } });
+
+    await waitFor(() => {
+      expect(
+        panelContainer.querySelector<HTMLElement>('.aa-Panel')
+      ).toBeInTheDocument();
+    });
+  });
+
+  test('retrieves the custom createElement from the renderer', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+    const mockCreateElement = jest.fn().mockImplementation(preactCreateElement);
+
+    document.body.appendChild(panelContainer);
+    autocomplete<{ label: string }>({
+      container,
+      panelContainer,
+      openOnFocus: true,
+      getSources() {
+        return [
+          {
+            sourceId: 'testSource',
+            getItems() {
+              return [{ label: '1' }];
+            },
+            templates: {
+              item({ item }) {
+                return item.label;
+              },
+            },
+          },
+        ];
+      },
+      render({ createElement }, root) {
+        expect(createElement).toBe(mockCreateElement);
+        render(createElement('div', null, 'testSource'), root);
+      },
+      renderer: {
+        createElement: mockCreateElement,
+        Fragment: PreactFragment,
+      },
+    });
+
+    const input = container.querySelector<HTMLInputElement>('.aa-Input');
+
+    fireEvent.input(input, { target: { value: 'a' } });
+
+    await waitFor(() => {
+      expect(
+        panelContainer.querySelector<HTMLElement>('.aa-Panel')
+      ).toBeInTheDocument();
+    });
+  });
+
+  test('retrieves the custom Fragment from the renderer', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+
+    const customFragment = (props: any) => props.children;
```

Comment:
```suggestion
    const CustomFragment = (props: any) => props.children;
```

Reply:
Done in https://github.com/algolia/autocomplete/pull/494/commits/0e2b1f30f70db56d70f5d36bd23aad2d853c7fce